include/Gaffer/Animation.h

```diff
@@ -414,6 +442,22 @@ class GAFFER_API Animation : public ComputeNode
 				/// iterator to end of range of active keys. (const access)
 				ConstKeyIterator end() const;
 
+				/// Get in extrapolation.
+				Extrapolation getExtrapolationIn() const;
+				/// Get out extrapolation.
+				Extrapolation getExtrapolationOut() const;
+				/// Get extrapolation in specified direction.
+				Extrapolation getExtrapolation( Animation::Direction direction ) const;
+				/// Set in extrapolation.
+				/// \undoable
+				void setExtrapolationIn( Extrapolation extrapolation );
+				/// Set out extrapolation.
+				/// \undoable
+				void setExtrapolationOut( Extrapolation extrapolation );
+				/// Set extrapolation in specified direction.
+				/// \undoable
+				void setExtrapolation( Animation::Direction direction, Extrapolation extrapolation );
```

Comment:
Feels like just these two alone would have us pretty well covered, and slimming down the number of functions would make it easier to understand/use :

```
Extrapolation getExtrapolation( Animation::Direction direction ) const;
void setExtrapolation( Animation::Direction direction, Extrapolation extrapolation );
```

Reply:
The extra functions are just there for convenience, `key.getExtrapolationIn()` is alot less to type and more concise to read than `key.getExtrapolation( Gaffer.Animation.Direction.In )`

Reply:
My thinking was that extrapolation is probably dealt with infrequently enough that the smaller API is a win overall.

Reply:
`setExtrapolationIn()` and `setExtrapolationOut()` are used by the serialisation to set the extrapolation mode of a curve. Using the more concise naming would result in less verbosity and therefore more readable and smaller scene files.

Reply:
`[get|set]Extrapolation[In|Out]()` convenience functions have now been removed. serialisation uses the more verbose API.